src/core/settings.hpp

```diff
@@ -91,28 +99,21 @@ class Q_MAPLIBRE_CORE_EXPORT Settings {
     std::function<std::string(const std::string &)> resourceTransform() const;
     void setResourceTransform(const std::function<std::string(const std::string &)> &);
 
-    void resetToTemplate(SettingsTemplate);
+    void setProviderTemplate(ProviderTemplate);
+    void setStyles(const Styles &styles);
+
+    const Styles &styles() const;
+    Styles providerStyles() const;
 
-    QVector<QPair<QString, QString>> defaultStyles() const;
+    Coordinate defaultCoordinate() const;
+    void setDefaultCoordinate(const Coordinate &);
+    double defaultZoom() const;
+    void setDefaultZoom(double);
 
-    mbgl::TileServerOptions *tileServerOptionsInternal() const;
+    mbgl::TileServerOptions *tileServerOptions() const;
 
 private:
-    GLContextMode m_contextMode;
-    MapMode m_mapMode;
-    ConstrainMode m_constrainMode;
-    ViewportMode m_viewportMode;
-
-    unsigned m_cacheMaximumSize;
-    QString m_cacheDatabasePath;
-    QString m_assetPath;
-    QString m_apiKey;
-    QString m_localFontFamily;
-    QString m_clientName;
-    QString m_clientVersion;
-    std::function<std::string(const std::string &)> m_resourceTransform;
-
-    mbgl::TileServerOptions *m_tileServerOptionsInternal{};
+    SettingsPrivate *d_ptr;
```

Comment:
Please use a smart pointer.

Reply:
I did a research and using PIMPL idiom rarely uses smart pointers.

Reply:
Maybe because there is lots of old pre-smart pointer code?

The C++ core guidelines do use a unique_ptr with PIMPL. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ri-pimpl

Reply:
OK, can we merge this now then and move everything to smart pointers in a separate MR?

Reply:
Sure.